pkgs/development/compilers/crystal/default.nix

```diff
@@ -95,16 +96,27 @@ let
       "--single-module" # needed for deterministic builds
     ];
 
+    # This makes sure we don't keep depending on the previous version of
+    # crystal used to build this one.
+    CRYSTAL_LIBRARY_PATH = "${placeholder "out"}/lib/crystal";
+
     # We *have* to add `which` to the PATH or crystal is unable to build stuff
     # later if which is not available.
     installPhase = ''
       runHook preInstall
 
       install -Dm755 .build/crystal $out/bin/crystal
       wrapProgram $out/bin/crystal \
-          --suffix PATH : ${lib.makeBinPath [ clang which ]} \
+          --suffix PATH : ${lib.makeBinPath [ pkgconfig clang which ]} \
           --suffix CRYSTAL_PATH : lib:$out/lib/crystal \
-          --suffix LIBRARY_PATH : ${lib.makeLibraryPath buildInputs}
+          --suffix CRYSTAL_LIBRARY_PATH : ${lib.makeLibraryPath (compiler.buildInputs ++ [
```

Comment:
how about a:
```
(compilter.buildInputs ++ (map lib.getLib [ pcre zlib .... ])
```
to avoid having to hardcode the output?

Reply:
Ah, I didn't know about this. I tried the `make*Path` functions, but none really fit well.

Reply:
OK, had to work around `buildInputs` turning into a list of `dev` outputs, so I moved that list outside. It seems a lot cleaner now.